### ISSUE TITLE
Wait condition times out when creating in an existing vpc on private subnet

### ISSUE THREAD

**Issue by joncourt** [aws-quickstart/quickstart-confluent-kafka#54]
I'm attempting to use this quickstart into a current vpc which we have a 'direct connect' connection to - through a virtual private gateway. Being essentially internal to our network the subnets are generally private - but do have a NAT for accessing the internet. The instances in the private subnets are available from our internal network.

I believe this may be causing a problem with the ClusterInfoCondition in that CloudFormation can't access the instance to retrieve details about it - looking briefly at the code i think this is what the templates and scripts are trying to do.

Is there a nice way to route CF via a bastion host or similar, or am i missing a trick in my config. I am able to use the quickstart successfully when i use the 'new vpc'.

Regards,
Jon

```
  | 13:05:25 UTC+1300 | CREATE_FAILED | AWS::CloudFormation::Stack | loyalty-kafka-stack | The following resource(s) failed to create: [ClusterInfoCondition].
-- | -- | -- | -- | -- | --
  | Physical ID:arn:aws:cloudformation:ap-southeast-2:817831548043:stack/loyalty-kafka-stack/efdc2690-c34d-11e7-9100-503f237b780a
  | 13:05:24 UTC+1300 | CREATE_FAILED | AWS::CloudFormation::WaitCondition | ClusterInfoCondition | WaitCondition timed out. Received 0 conditions when expecting 1
  | Physical ID:arn:aws:cloudformation:ap-southeast-2:817831548043:stack/loyalty-kafka-stack/efdc2690-c34d-11e7-9100-503f237b780a/ClusterInfoHandle
```

Config:
```bash
       --parameters ParameterKey=AssignPublicIP,ParameterValue=false,UsePreviousValue=false \
                 ParameterKey=BootDiskSize,ParameterValue=24,UsePreviousValue=false \
                 ParameterKey=BrokerNodeInstanceType,ParameterValue=m3.large,UsePreviousValue=false \
                 ParameterKey=BrokerNodeSpotPrice,ParameterValue=0.00,UsePreviousValue=false \
                 ParameterKey=BrokerNodeStorage,ParameterValue=512,UsePreviousValue=false \
                 ParameterKey=BrokerNodeStorageType,ParameterValue=st1,UsePreviousValue=false \
                 ParameterKey=ClusterName,ParameterValue=loyalty-kafka-cluster,UsePreviousValue=false \
                 ParameterKey=ConfluentEdition,ParameterValue="Confluent Open Source",UsePreviousValue=false \
                 ParameterKey=ConfluentVersion,ParameterValue=3.3.0,UsePreviousValue=false \
                 ParameterKey=ConnectorURLs,ParameterValue=,UsePreviousValue=false \
                 ParameterKey=KeyPairName,ParameterValue=loyalty-kafka-keypair,UsePreviousValue=false \
                 ParameterKey=LinuxOSAMI,ParameterValue=Amazon-Linux-HVM,UsePreviousValue=false \
                 ParameterKey=NumBrokers,ParameterValue=3,UsePreviousValue=false \
                 ParameterKey=NumWorkers,ParameterValue=2,UsePreviousValue=false \
                 ParameterKey=NumZookeepers,ParameterValue=0,UsePreviousValue=false \
                 ParameterKey=RemoteAccessCIDR,ParameterValue=10.0.0.0/8,UsePreviousValue=false \
                 ParameterKey=SSHAccessCIDR,ParameterValue=10.0.0.0/8,UsePreviousValue=false \
                 ParameterKey=SubnetID,ParameterValue=subnet-54f0c422\\,subnet-7d727419,UsePreviousValue=false \
                 ParameterKey=VPCID,ParameterValue=vpc-f08e9395,UsePreviousValue=false \
                 ParameterKey=WorkerNodeInstanceType,ParameterValue=m3.large,UsePreviousValue=false \
                 ParameterKey=WorkerNodeSpotPrice,ParameterValue=0.00,UsePreviousValue=false \
                 ParameterKey=WorkerNodeStorage,ParameterValue=0,UsePreviousValue=false \
                 ParameterKey=ZookeeperNodeInstanceType,ParameterValue=m3.large,UsePreviousValue=false \
                 ParameterKey=ZookeeperNodeSpotPrice,ParameterValue=0.00,UsePreviousValue=false \
                 ParameterKey=ZookeeperNodeStorage,ParameterValue=0,UsePreviousValue=false
```

**Comment by joncourt** [comment]
Actually i believe i've solved this on a fork. The issue i found was my instances (only tested a broker one though) can't resolve 'instance-data' in the murl variables. I modified them all to the IP recommended by AWS as per http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html (i.e. 169.254.169.254) and now the quickstart is succeeding.

Was the hosts file supposed to be set to map this ip to instance-data in the AMI perhaps and no longer does?

This IP is stable I think, AWS recommends using the raw IP according to mattj@aws.com on this thread: https://forums.aws.amazon.com/message.jspa?messageID=536813.

I've made the changes and will create a pull request for you to merge if you whish.
